plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMapping.java

```diff
@@ -70,6 +74,9 @@ public S3SecurityMapping(
         this.allowedIamRoles = requireNonNull(allowedIamRoles, "allowedIamRoles is null")
                 .orElse(ImmutableList.of());
 
+        this.kmsKeyId = requireNonNull(kmsKeyId, "kmsKeyId is null");
+        this.sseKmsKeyId = requireNonNull(sseKmsKeyId, "sseKmsKeyId is null");
```

Comment:
can `kmsKeyId` be used together `sseKmsKeyId`? Can we only set only one of them? Do we need any more validation to prevent configuring something that is invalid for AWS?

Reply:
> can kmsKeyId be used together sseKmsKeyId? Can we only set only one of them? Do we need any more validation to prevent configuring something that is invalid for AWS?

These are good points. I don't know. For global configuration we do not have any (much) validation in place. And one can easily provide a configuration where some of the provided parameters would be ignored (e.g `hive.s3.sse.type=S3` and `hive.s3.sse.kms-key-id=XXXX`). I did not extend any validation as part of this PR as it was not the goal. Yet I agree that having better understanding of what makes sense, and validating against that would make sense.

> sseKmsKeyId is only used when writing, and only when sseEnabled is enabled. It wouldn't make sense to allow setting it conditionally if SSE is not enabled. Would we want this to also enable SSE?

Good point. We should either:
* validate that SSE is enabled globally and it is using KMS mode.
* OR force SSE enabled and KMS mode if there is sseKmsKeyId set in the mapping.

I am more leaning towards the former. Not sure what should be the action when we see that `sseKmsKeyId` set in mapping does not make sense as it would not be used anyway, due to global settings. Should we fail the request or just not set the `sseKmsKeyId` in the final config? Or maybe we should perform validation of whole mapping file at Trino startup? I guess no, because it does not play well with periodic file reloading.